### DISCUSSION TITLE
Absolute path to file of my project

### DISCUSSION THREAD

**Discussion by marcperrinp** [vercel/vercel#4614] [Help]
Hi,

In my code, I fork the main node process into a child process using

```
import { fork } from 'child_process'

const child = fork('some-script.js')
```

It works great locally, but not in production because "some-script.js" is not found at path `/var/task/some-script.js`. My guess is that locally I run node from my project's root folder, whereas in Vercel, node is ran from somewhere else.

My question is: What is the absolute path to my "some-script.js" file (knowing that it sits at the root of my project)?

I have already tried `__dirname`, `process.cwd` and `__filename`.

Thank you to anyone for their help!

**Comment by marcperrinp** [comment]
I guess my question boils down to: Is it possible to host a file in Vercel outside of a Serverless function?

![image](https://user-images.githubusercontent.com/48444523/84256082-c705dd00-ab13-11ea-89c1-aef17c074e84.png)

As I said in another question, I'm new to Vercel, so this might not be the way it's meant to be used. But that would be a bummer, because it's the last piece of the puzzle for me. I just need access to a file (which is not a serverless function, meaning there is no export of (req, res)) as said above. I hope I can find a solution to this :/

**Comment by mcsdevv** [comment]
Hey Marc,

The easiest way to do this is to place the file inside of the `/public` directory, does this work for your use case?

As a heads up, I've sent the same reply to your ticket also - please let me know the best place for you to discuss this 👍

**Reply by marcperrinp** [reply]
What would be the path to access a file in the `/public` directory?

**Comment by marcperrinp** [comment]
Here is far better, I didn't know about this forum before 👍 

About your suggestion: the `/public` folder is public though, right? I wouldn't be thrilled about that, but there is no sensitive data in it, so I guess it could do the trick!

However, I would I be able to access is in the file system? (N.B: as you see on the above snippet, it needs to be accessed from the file system and not by an http call, so https://my-domain/public/myscript.js won't work)

**Comment by styfle** [comment]
Try the following:

```js
// File is api/example.js
const { fork } = require('child_process')
const { join } = require ('path')

const child = fork(join(__dirname, '../child.js'))
```

That being said, I think forking a child process is an anti-pattern for serverless.

Think of the parent process as the http client and the child process as the serverless function. "Fork" should really be "HTTP GET" and the function itself should not be running multiple processes. As the traffic increases, the function will scale accordingly automatically.

**Reply by marcperrinp** [reply]
Thanks but that does not work :( As I said in my original post:

> I have already tried __dirname, process.cwd and __filename.

I tried with child.js at the root of my project and in the public directory, as suggested by @mcsdevv 

Isn't there a way to hard code something like `fork('/var/www/my-project-name/child.js')` ?

**Reply by joefitter** [reply]
did you ever find an answer to this? I'm coming up against the exact same issue as I can't locate my worker

**Reply by BossBele** [reply]
Having the same issue

**Reply by pdhoward** [reply]
same

**Reply by danielavalos1** [reply]
same

**Comment by pdhoward** [comment]
Here is what works for me:

1.  upgraded to next v14.1.4+
2. configured paths for test and production as follows:
  // Resolve must be used to Production to set correct path
    const basePathSegment = `repos/${username}/${repoName}/tests`
    const fullPath = path.join(basePathSegment, subdir, file);
    const testScriptPath = isProduction ? resolve(fullPath) : fullPath;   

Note that path.resolve is required to dynamically set the path in vercel environment
